### PR TITLE
Porting univ3 methods and add script to min lowest range

### PR DESCRIPTION
**Pull request by petrovska-petro**
Tackles #32 

Command:

```
brownie run  scripts/issue/32/mint_lowest_range_uni_v3
```

### CODE REVIEW

**Comment by petrovska-petro** [comment]
Reminder: thinking to add another wd method, which i think it can allow to go directly from `crvIbBTC` to `wbtc`

**Comment by petrovska-petro** [comment]
Thru zapper seems to behave well, avoiding slippage issue:

Snap locally:

```
snapshot result for 0x042B32Ac6b453485e357938bdC38e0340d4b9276:
                         balance_before         balance_after           balance_delta
symbol                                                                               
bibbtc/sbtcCRV-f  25.000000000000000000  0.000000000000000000  -25.000000000000000000
WBTC               0.046680780000000000  5.560590170000000000    5.513909390000000000 
```

File: great_ape_safe/ape_api/uni_v3.py [hunk 1]
```diff
@@ -18,33 +35,357 @@ def __init__(self, safe):
...
+        """
+        It will decrease the liquidity from a specific NFT
+        and collect the fees earned on it
+        optional: to completly burn the NFT
+        """
+        position = self.nonfungible_position_manager.positions(token_id)
+        deadline = chain.time() + self.deadline
+
+        liquidity = position["liquidity"]
+        sqrtRatioX96, _, _, _, _, _, _ = self.v3pool_wbtc_badger.slot0()
```
Comment (jorijnsmit) [inline great_ape_safe/ape_api/uni_v3.py:51]: this should come from `pool` no?

**Review by jorijnsmit — COMMENTED** [review]
so many uni v3 guns!

just found one (old) reference to v3pool_wbtc_badger.

File: great_ape_safe/ape_api/curve.py [hunk 2]
```diff
@@ -87,6 +85,35 @@ def _get_n_coins(self, lp_token):
...
+            pass
+        else:
+            registry = self._get_registry(lp_token)
+            pool = self._get_pool_from_lp_token(lp_token)
+            if registry == self.metapool_registry:
+                return registry.get_n_coins(pool)
+            else:
+                return registry.get_n_coins(pool)[
+                    0
+                ]  # note [1] tells us if there is a wrapped coin!
```
Comment (jalbrekt85) [inline great_ape_safe/ape_api/curve.py:116]: Not sure what other changes there are for `get_n_coins`besides the abscence of the metapool registry between v1/v2 but this worked for me
```suggestion
def _get_n_coins(self, lp_token):
        registry = self._get_registry(lp_token)
        pool = self._get_pool_from_lp_token(lp_token)
        if not self.is_v2 and registry == self.metapool_registry:
            return registry.get_n_coins(pool)
        else:
            return registry.get_n_coins(pool)[0]
```

**Review by jalbrekt85 — COMMENTED** [review]
zapper method if very helpful. I added a suggestion for getting `_get_n_coins` to work with v2. Also this method seems to duplicated on L76 and 104

File: great_ape_safe/ape_api/uni_v3.py (hunk 1 above)
Comment (petrovska-petro) [reply great_ape_safe/ape_api/uni_v3.py]: sorted in last commit

**Review by jalbrekt85 — APPROVED** [review]
looks good

**Pull request merged**